### PR TITLE
xcbuild 0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by steeve**
Signed-off-by: Steeve Morin <steeve.morin@gmail.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is an updated PR from #5986.

### CODE REVIEW

**Comment by woodruffw** [comment]
```
Error: 5 problems in 1 formula
xcbuild:
  * cmake dependency should be
      depends_on "cmake" => :build
    Or if it is indeed a runtime dependency
      depends_on "cmake" => :run
  * pkg-config dependency should be
      depends_on "pkg-config" => :build
    Or if it is indeed a runtime dependency
      depends_on "pkg-config" => :run
  * cmake dependency should be
      depends_on "cmake" => :build
    Or if it is indeed a runtime dependency
      depends_on "cmake" => :run
  * pkg-config dependency should be
      depends_on "pkg-config" => :build
    Or if it is indeed a runtime dependency
      depends_on "pkg-config" => :run
  * The installation was broken.
    Broken dylib links found:
      /usr/lib/libcompression.dylib
```

File: Formula/xcbuild.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on "pkg-config"
+
+  def install
+    system "cmake", ".", "-G", "Ninja", *std_cmake_args
+    system "ninja", "install"
+    bin.install_symlink "#{prefix}/usr/bin/xcbuild"
+  end
+
+  test do
+    system "#{bin}/xcbuild", "-showsdks"
```
Comment (woodruffw) [inline Formula/xcbuild.rb]: Can we test the output of this with `assert_match` or `assert_equals`?

File: Formula/xcbuild.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Xcbuild < Formula
+  desc "Xcode-compatible build tool"
+  homepage "https://github.com/facebook/xcbuild"
+  url "https://github.com/facebook/xcbuild.git", :tag => "0.1.0", :revision => "08575f65f1c907e280ed4663b5461b7cfaf5cfaa"
```
Comment (woodruffw) [inline Formula/xcbuild.rb]: Break this up into multiple lines, one per parameter.

Comment (woodruffw) [inline Formula/xcbuild.rb]: Or better yet, use [the release](https://github.com/facebook/xcbuild/releases/tag/0.1.0) instead. We use tags only when releases aren't available.

File: Formula/xcbuild.rb (hunk 1 above)
Comment (steeve) [inline Formula/xcbuild.rb]: Not really, as it is dependant on what SDKs the user has installed on their machine.

For instance mine is:
```
$ xcbuild -showsdks
iOS SDKs:
	iOS 10.2                        -sdk iphoneos10.2
	iOS 10.2                        -sdk iphoneos10.2

iOS Simulator SDKs:
	Simulator - iOS 10.2            -sdk iphonesimulator10.2
	Simulator - iOS 10.2            -sdk iphonesimulator10.2

macOS SDKs:
	macOS 10.12                     -sdk macosx10.12
	macOS 10.12                     -sdk macosx10.12

tvOS SDKs:
	tvOS 10.1                       -sdk appletvos10.1
	tvOS 10.1                       -sdk appletvos10.1

tvOS Simulator SDKs:
	Simulator - tvOS 10.1           -sdk appletvsimulator10.1
	Simulator - tvOS 10.1           -sdk appletvsimulator10.1

watchOS SDKs:
	watchOS 3.1                     -sdk watchos3.1
	watchOS 3.1                     -sdk watchos3.1

watchOS Simulator SDKs:
	Simulator - watchOS 3.1         -sdk watchsimulator3.1
	Simulator - watchOS 3.1         -sdk watchsimulator3.1```

**Comment by woodruffw** [comment]
Looks like this requires `googletest` as a resource. 

```
CMake Error at CMakeLists.txt:54 (add_subdirectory):
  add_subdirectory given source "ThirdParty/googletest/googletest" which is
  not an existing directory.
```

**Comment by steeve** [comment]
Is it pulled as a git submodule, and by using the release, it is not there, I think we should revert back to the tag.

**Comment by steeve** [comment]
Sorry I did not think of this side effect when switching to the release tgz.

File: Formula/xcbuild.rb (hunk 2 above)
Comment (steeve) [inline Formula/xcbuild.rb]: Should I indent those lines to differentiate them from the rest ?

**Comment by ilovezfs** [comment]
https://github.com/facebook/xcbuild/blob/master/.gitmodules
```
[submodule "ThirdParty/googletest"]
	path = ThirdParty/googletest
	url = https://github.com/google/googletest.git
[submodule "ThirdParty/linenoise"]
	path = ThirdParty/linenoise
	url = https://github.com/antirez/linenoise.git
```

Seems weird that linenoise isn't revision pegged.

**Comment by woodruffw** [comment]
> Sorry I did not think of this side effect when switching to the release tgz.

Not your fault, nobody foresaw it :smile:

**Comment by steeve** [comment]
Updated the PR to switch back to the tag

File: Formula/xcbuild.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Xcbuild < Formula
+  desc "Xcode-compatible build tool"
+  homepage "https://github.com/facebook/xcbuild"
+  url "https://github.com/facebook/xcbuild.git",
+    :tag => "0.1.0",
```
Comment (ilovezfs) [inline Formula/xcbuild.rb]: please indent this so the `:` is under the `"`

File: Formula/xcbuild.rb (hunk 2 above)
Comment (woodruffw) [inline Formula/xcbuild.rb]: Yep. See `docker` for an example.

File: Formula/xcbuild.rb (hunk 3 above)
Comment (steeve) [inline Formula/xcbuild.rb]: Sure

Comment (steeve) [inline Formula/xcbuild.rb]: I am not sure wether it will clone the submodules, will it ?

Comment (ilovezfs) [inline Formula/xcbuild.rb]: yeah that's what happens if you you use the tag.

**Comment by MikeMcQuaid** [comment]
A weird bad link on Yosemite: https://bot.brew.sh/job/Homebrew%20Core/14289/version=yosemite/console

**Comment by fxcoudert** [comment]
For what it's worth, there is now a [0.1.1 release](https://github.com/facebook/xcbuild/releases/tag/0.1.1), maybe it would build better?

**Comment by fxcoudert** [comment]
Closing for now, please reopen once it is fixed.

**Pull request closed without merging**